src/frontend/packages/lib_video/src/components/common/VideoWidgetProvider/WidgetsContainer/index.tsx

```diff
@@ -21,25 +21,39 @@ interface WidgetsContainerProps {
 
 export const WidgetsContainer = ({ children }: WidgetsContainerProps) => {
   const [infoWidgetModal, setInfoWidgetModal] = useInfoWidgetModal();
+  const mainContainerRef = useRef<HTMLDivElement>(null);
+  const [nbrOfColumns, setNbrOfColumns] = useState(2);
 
-  const layoutSize = React.useContext(ResponsiveContext);
-  let nbrOfColumns: number;
-  switch (layoutSize) {
-    case 'small':
-      nbrOfColumns = 1;
-      break;
-    case 'medium':
-      nbrOfColumns = 2;
-      break;
-    default:
-      nbrOfColumns = 3;
-  }
-  const mapper: unknown[] = Array(nbrOfColumns).fill(0);
+  useEffect(() => {
+    const handleResize = () => {
+      if (!mainContainerRef.current?.offsetWidth) {
+        return;
+      }
+
+      const minWidgetWidth = 380;
+      let columns = Math.floor(
+        mainContainerRef.current?.offsetWidth / minWidgetWidth,
+      );
+      columns = columns > 3 ? 3 : columns;
+      setNbrOfColumns(columns || 1);
```

Comment:
Maybe we could extract those values to conf/global.ts ?
Something like `MAX_WIDGETS_COLUMNS` and `MIN_WIDGETS_COLUMNS` ?

Reply:
I added some globals to be cleaner. 
https://github.com/openfun/marsha/pull/2086/commits/9226d813cafb0b1748d8aa0b2cfab4bd7e2dbf42

I don't know about `conf/global.ts`, if it is used only in this module IMO I would rather let it inside the module. However if it is used in other modules, yes it would make sense to add them in `conf/global.ts` .